nixos/modules/system/boot/systemd/sysupdate.nix

```diff
@@ -0,0 +1,87 @@
+{ config, lib, pkgs, ... }:
+
+let
+  cfg = config.systemd.sysupdate;
+
+  writeDefinition = name: transferDefinition: pkgs.writeText
+    "${name}.conf"
+    (lib.generators.toINI { } transferDefinition);
+
+  listOfDefinitions = lib.mapAttrsToList
+    writeDefinition
+    (lib.filterAttrs (k: _: !(lib.hasPrefix "_" k)) cfg.transfers);
+
+  definitionsDirectory = pkgs.runCommand "sysupdate.d" { } ''
+    mkdir -p $out
+    ${(lib.concatStringsSep "\n"
+      (map (pkg: "cp ${pkg} $out/${pkg.name}") listOfDefinitions)
+    )}
+  '';
+in
+{
+  options.systemd.sysupdate = {
+
+    enable = lib.mkEnableOption (lib.mdDoc "systemd-sysupdate") // {
+      description = lib.mdDoc ''
+        Atomically update the host OS, container images, portable service
+        images or other sources.
```

Comment:
From the description it is not clear what gets enabled when I turn on this option.

Reply:
Yeah, I'm also missing some more documentation about how this should work, and what the test is testing.